src/Aspire.Hosting/DistributedApplicationBuilder.cs

```diff
@@ -245,6 +245,7 @@ public DistributedApplicationBuilder(DistributedApplicationOptions options)
         _innerBuilder.Services.AddSingleton<ResourceNotificationService>();
         _innerBuilder.Services.AddSingleton<ResourceLoggerService>();
         _innerBuilder.Services.AddSingleton<ResourceCommandService>(s => new ResourceCommandService(s.GetRequiredService<ResourceNotificationService>(), s.GetRequiredService<ResourceLoggerService>(), s));
+        _innerBuilder.Services.AddSingleton<ContainerExecService>(s => new ContainerExecService(s.GetRequiredService<ResourceNotificationService>(), s.GetRequiredService<ResourceLoggerService>(), s.GetRequiredService<IDcpExecutor>(), s.GetRequiredService<DcpNameGenerator>()));
```

Comment:
Why not just register the singleton - you shouldn't need the callback to construct the type here.

Reply:
changed registration